sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/percentiles.scala

```diff
@@ -154,12 +155,7 @@ abstract class PercentileBase
       return Seq.empty
     }
 
-    val ordering = child.dataType match {
-      case numericType: NumericType => numericType.ordering
-      case intervalType: YearMonthIntervalType => intervalType.ordering
-      case intervalType: DayTimeIntervalType => intervalType.ordering
-      case otherType => QueryExecutionErrors.unsupportedTypeError(otherType)
-    }
+    val ordering = PhysicalDataType.ordering(child.dataType)
```

Comment:
what's the difference between this and `PhysicalDataType(child.dataType).ordering`?

Reply:
`PhysicalDataType(child.dataType).ordering?` returns `Ordering[InternalType]`

`PhysicalDataType.ordering(child.dataType)` returns `Ordering[Any]`.

I found the former caused many compilation complains and eventually it become the latter.

Reply:
does any place need `Ordering[InternalType]`? if not we can always return `Ordering[Any]`

Reply:
Can we try to drop `InternalType` in the last thus try to make `Ordering` and all others to just be `ANY`?

Right now I still expect to move `Numeric`, `Integral` etc. which looks like use `InternalType`. After smoothly moving those things I plan to drop `InternalType` at once and make everything `ANY`. However now it is more easier to keep the original code as much as possible.